ci/jenkins/templates/e2e-dev-template.yaml

```diff
@@ -48,7 +52,7 @@
                 credentials-id: '{repo-credentials}'
                 refspec: '+refs/heads/*:refs/remotes/origin/* +refs/pull/*/head:refs/remotes/origin/pr/*'
                 branches:
-                    - '$BRANCH'
+                    - '$JENKINSFILE_BRANCH'
```

Comment:
I'm not sure this will work. I was thinking about defining here only JENKINSFILE_BRANCH (needed to pick the Jenkinsfile at the start of the job) and use the BRANCH in the SCM step in the pipeline.

Reply:
it does, https://ci.suse.de/view/CaaSP/view/v4.5%20Dashboard/job/caasp-jobs/job/conformance/job/v4.5/job/vmware/136/console tested with `non-existant-branch` and it clones the repo and tries to change to that branch to pick the jenkinsfile:

```
Checking out git https://github.com/SUSE/skuba.git into /var/lib/jenkins/jobs/caasp-jobs/jobs/conformance/jobs/v4.5/jobs/vmware/workspace@script to read ci/jenkins/pipelines/skuba-conformance.Jenkinsfile
using credential github-token
Wiping out workspace first.
Cloning the remote Git repository
Cloning repository https://github.com/SUSE/skuba.git
 > git init /var/lib/jenkins/jobs/caasp-jobs/jobs/conformance/jobs/v4.5/jobs/vmware/workspace@script # timeout=10
Fetching upstream changes from https://github.com/SUSE/skuba.git
 > git --version # timeout=10
 > git --version # 'git version 2.26.2'
using GIT_ASKPASS to set credentials 
 > git fetch --tags --force --progress -- https://github.com/SUSE/skuba.git +refs/heads/*:refs/remotes/origin/* # timeout=10

 > git config remote.origin.url https://github.com/SUSE/skuba.git # timeout=10
 > git config --add remote.origin.fetch +refs/heads/*:refs/remotes/origin/* # timeout=10
 > git config remote.origin.url https://github.com/SUSE/skuba.git # timeout=10
Fetching upstream changes from https://github.com/SUSE/skuba.git
using GIT_ASKPASS to set credentials 
 > git fetch --tags --force --progress -- https://github.com/SUSE/skuba.git +refs/heads/*:refs/remotes/origin/* +refs/pull/*/head:refs/remotes/origin/pr/* # timeout=10

 > git rev-parse origin/non_existant^{commit} # timeout=10
 > git rev-parse non_existant^{commit} # timeout=10
ERROR: Couldn't find any revision to build. Verify the repository and branch configuration for this job.
ERROR: Maximum checkout retry attempts reached, aborting
Finished: FAILURE
```

Reply:
Manually changed the job BTW to add the above patch, its now reverted!

Reply:
oh, I misread the change! I was thinking you had both branches in the configuration!  Sorry for the noise.